src/main/java/com/sprint/mission/discodeit/service/basic/BasicUserService.java

```diff
@@ -0,0 +1,93 @@
+import com.sprint.mission.discodeit.DTO.User.FindUserDTO;
+import com.sprint.mission.discodeit.DTO.User.UpdateUserDTO;
+import com.sprint.mission.discodeit.entity.User;
+import com.sprint.mission.discodeit.entity.UserStatus;
+import com.sprint.mission.discodeit.repository.BinaryContentRepository;
+import com.sprint.mission.discodeit.repository.UserRepository;
+import com.sprint.mission.discodeit.repository.UserStatusRepository;
+import com.sprint.mission.discodeit.service.UserService;
+import com.sun.jdi.request.DuplicateRequestException;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+@Service
+@RequiredArgsConstructor
+public class BasicUserService implements UserService{
+    private final UserRepository userRepository;
+    private final UserStatusRepository userStatusRepository;
+    private final BinaryContentRepository binaryContentRepository;
+
+    @Override
+    public User create(CreateUserDTO createUserDTO){
+        List<User> users = userRepository.findAll();
+        boolean nameIsDuplicate = users.stream()
+                .anyMatch(user -> user.getUsername().equals(createUserDTO.userName()));
+
+        if(nameIsDuplicate){
+            throw new DuplicateRequestException("Username already exists");
+        }
+
+        boolean emailIsDuplicate = users.stream()
+                .anyMatch(user -> user.getEmail().equals(createUserDTO.userName()));
+
+        if(emailIsDuplicate){
+            throw new DuplicateRequestException("Email already exists");
+        }
+
+        User user = userRepository.save(new User(createUserDTO));
+        userStatusRepository.save(new UserStatus(user.getId()));
+        return user;
+    }
+
+    @Override
+    public FindUserDTO find(UUID id) {
+        User user = userRepository.findById(id).orElseThrow(()->new NullPointerException("User not found"));
+        UserStatus userStatus = userStatusRepository.findByUserId(user.getId()).orElseThrow(()->new NullPointerException("UserStatus not found"));
+        return new FindUserDTO(user,userStatus);
+    }
+
+    @Override
+    public FindUserDTO findEmail(String email) {
+        User user = userRepository.findByEmail(email).orElseThrow(()->new NullPointerException("User not found"));
+        UserStatus userStatus = userStatusRepository.findByUserId(user.getId()).orElseThrow(()->new NullPointerException("UserStatus not found"));
+        return new FindUserDTO(user,userStatus);
+    }
+
+    @Override
+    public List<FindUserDTO> findAll() {
+        List<User> users = userRepository.findAll();
+        List<UserStatus> userStatuses = userStatusRepository.findAll();
+        Map<UUID, UserStatus> userStatusMap = userStatuses.stream()
+                .collect(Collectors.toMap(UserStatus::getUserId, userStatus->userStatus));
```

Comment:
Java8에 등장한 [Function.identity()](https://docs.oracle.com/javase/8/docs/api/java/util/function/Function.html#identity--)을 사용하는게 관용적일 것 같요.

기능상의 차이는 없지만 Function.identity()이 Java8에서 제공하는 표준방법이라 의도가 좀더 명확해서요!

```suggestion
        Map<UUID, UserStatus> userStatusMap = userStatuses.stream()
                .collect(Collectors.toMap(UserStatus::getUserId, Function.identity()));
```